gpii/node_modules/singleInstance/test/SingleInstanceTests.js

```diff
@@ -0,0 +1,207 @@
+ * Federal Government.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("universal"),
+    fs = require("fs"),
+    os = require("os"),
+    child_process = require("child_process");
+
+var jqUnit = fluid.require("node-jqunit");
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.tests.singleInstance");
+
+require("singleInstance");
+
+jqUnit.module("gpii.tests.singleInstance");
+
+fluid.defaults("gpii.tests.singleInstance.testData", {
+    gradeNames: "fluid.modelComponent",
+    model: {
+        tests: [
+            {
+                // no pidfile
+                testData: null,
+                expected: {
+                    checkInstance: null,
+                    registerInstance: true,
+                    file: "$processes.own.pid",
+                    deregisterInstance: false
+                }
+            },
+            {
+                // junk content
+                testData: "not a pid",
+                expected: {
+                    checkInstance: null,
+                    registerInstance: true,
+                    file: "$processes.own.pid",
+                    deregisterInstance: false
+                }
+            },
+            {
+                // non-running pid
+                testData: "-100",
+                expected: {
+                    checkInstance: null,
+                    registerInstance: true,
+                    file: "$processes.own.pid",
+                    deregisterInstance: false
+                }
+            },
+            {
+                // this process
+                testData: "$processes.own.pid",
+                expected: {
+                    checkInstance: "$processes.own.pid",
+                    registerInstance: true,
+                    file: "$processes.own.pid",
+                    deregisterInstance: true
+                }
+            },
+            {
+                // another process
+                testData: "$processes.other.pid",
+                expected: {
+                    checkInstance: "$processes.other.pid",
+                    // If the implementation was complete, then registerInstance should succeed because the test process
+                    // isn't another GPII instance.
+                    registerInstance: false,
+                    file: null,
+                    deregisterInstance: false
+                }
+            }
+        ],
+        processes: {
+
+        }
+    },
+    invokers: {
+        getTests: "gpii.tests.singleInstance.getTests({that}.model)"
+    }
+});
+
+/**
+ * Extracts the test items from model, performing look-ups from it ("$like.this")
+ *
+ * @param model {Object}
+ * @return {Array} The test items.
+ */
+gpii.tests.singleInstance.getTests = function (model) {
+    // Deeply handle "$lookup" values
+    var getRecursive = function (value) {
```

Comment:
Cleaner if this is its own top-level function rather than being embedded in another. It's ok for the driver to just say something like var expanded = gpii.tests.singleInstance.resolveReferences(gpii.tests.singleInstance.testDefs);